scripts/build-custom-app.sh

```
@@ -0,0 +1,241 @@
+cp -r "${themes_path}/cclw/components/LandingSearchForm.tsx" "${components_dir}/LandingSearchForm.tsx"
+printf "%s" "${HERO_CONTENT}" >"${components_dir}/Hero.tsx"
+
+touch "${components_dir}/MethodologyLink.tsx"
+echo "${METHODOLOGY_LINK_CONTENT}" >"${components_dir}/MethodologyLink.tsx"
+
+touch "${pages_dir}/homepage.tsx"
+echo "${HOMEPAGE_CONTENT}" >"${pages_dir}/homepage.tsx"
+
+touch "${layouts_dir}/main.tsx"
+echo "${MAIN_CONTENT}" >"${layouts_dir}/main.tsx"
+
+e2e_test_pages_dir="${e2e_test_dir}/pages"
+mkdir "${e2e_test_pages_dir}"
+touch "${e2e_test_pages_dir}/homepage.cy.js"
+echo "${E2E_TEST_PAGE_CONTENT}" >"${e2e_test_pages_dir}/homepage.cy.js"
+
+# Copy the tailwind.config.ts and styles.scss from the cclw theme directory
+cp -r "${themes_path}/cclw/tailwind.config.js" "${new_dir}/tailwind.config.js"
+printf "Copied tailwind.config.js file from %s to %s.\n\n" "${themes_path}/cclw" "${new_dir}"
+
+cp -r "${themes_path}/cclw/styles/styles.scss" "${styles_dir}/styles.scss"
+printf "Copied styles.scss file from %s to %s.\n\n" "${themes_path}/cclw" "${styles_dir}"
+
+touch "${new_dir}/redirects.json"
+echo "[]" >"${new_dir}/redirects.json"
+
+# Update the ci-cd.yml, tsconfig.json and types.ts files accordingly
+sed -e "s/theme: \[\(.*\)\]/theme: [\1, ${theme}]/" .github/workflows/ci-cd.yml >.github/workflows/ci-cd.yml.tmp && mv .github/workflows/ci-cd.yml.tmp .github/workflows/ci-cd.yml
+
+sed -e '/"paths":/,/}/{
+		/}/s/}/,\n    "@'"${theme}"'\/*": ["themes\/'"${theme}"'\/*"]\n  }/
+	}' "tsconfig.json" >"temp_tsconfig.json" && mv "temp_tsconfig.json" "tsconfig.json"
+
+sed -e "s/export type TTheme = /export type TTheme = \"${theme}\" | /" src/types/types.ts >src/types/types.ts.tmp && mv src/types/types.ts.tmp src/types/types.ts
+
+printf "\033[1;34mNow that's done, set the environment variable by running: export THEME=%s\n \033[0m\n" "${theme}"
+
+# Check trunk.io is installed
+if ! command -v trunk &>/dev/null; then
```

This should be done near the start so we can fail fast